src/xdg.c

```diff
@@ -271,18 +271,16 @@ lookup_view_by_xdg_toplevel(struct server *server,
 }
 
 static void
-position_xdg_toplevel_view(struct view *view)
+position_xdg_toplevel_view(struct view *view, int *x, int *y)
 {
 	struct wlr_xdg_surface *xdg_surface = xdg_surface_from_view(view);
 	struct wlr_xdg_toplevel *parent_xdg_toplevel =
 		xdg_toplevel_from_view(view)->parent;
 
 	if (!parent_xdg_toplevel) {
-		struct wlr_box box =
-			output_usable_area_from_cursor_coords(view->server);
-		view->current.x = box.x;
-		view->current.y = box.y;
-		view_center(view);
+		view_compute_centered_position(view,
+			output_from_cursor_coords(view->server),
+			view->pending.width, view->pending.height, x, y);
```

Comment:
Not completely thought though but anyway:

Could we keep using `view_center(view, output_from_cursor_coords(view->server))` here and move the `wlr_scene_node_set_enabled(&view->scene_tree->node, true)` after the `if (!view->been_mapped)` branch?

Then we could keep `view_compute_centerd_position()` private.

Reply:
Looking at it again, I think `view_center` could in fact work. It would call through to `view->impl->configure` unnecessarily, but since only the position (not size) has changed, that should be a no-op.

Not sure about moving the `wlr_scene_node_set_enabled` call. I haven't really looked at that as it didn't seem necessary to touch it for this fix.

Reply:
Delaying the scene node enable call would just mean that we only make the view visible after all in-between steps (e.g. calls to `view_moved()`) have been done. So we shouldn't see any in-between states on the screen that way.

Reply:
I see, makes sense. Would it be okay with you to defer that change to a future PR?